app/lib/test_executor.rb

```diff
@@ -6,26 +6,37 @@ def self.images
     @@images ||= {}
   end
 
-  def self.run_tests(submission)
-    problem = Problem.includes(:tests).find(submission.problem_id)
+  def self.correct_submission?(submission)
+    tests = Test.where(problem_id: submission.problem_id)
 
-    problem.tests.each do |test|
-      run_test(submission, test)
+    tests.each do |test|
```

Comment:
If we don't care that all tests run on the submission, we could do this, or
```
tests.all? do |test|
  correct_output?(submission, test)
end
```
as the return value.